src/Core/CustomMethods.php

```diff
@@ -285,7 +285,8 @@ protected function removeMethodsFrom($property, $index = null)
                     
                 $methodInfo = self::$extra_methods[$class][$method];
 
-                if ($methodInfo['property'] === $property && $methodInfo['index'] === $index) {
+                if (isset($methodInfo['property']) && $methodInfo['property'] === $property &&
+                    isset($methodInfo['index']) && $methodInfo['index'] === $index) {
```

Comment:
```suggestion
                    isset($methodInfo['index']) && $methodInfo['index'] === $index
                ) {
```

Minor formatting suggestion